dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionInputAttribute.cs

```diff
@@ -5,40 +5,44 @@
 namespace Microsoft.SemanticKernel.SkillDefinition;
 
 /// <summary>
-/// Attribute to describe the main parameter required by a native function,
-/// e.g. the first "string" parameter, if the function requires one.
+/// Attribute to describe the main "input" parameter required by a native function.
 /// </summary>
 /// <remarks>
-/// The class has no constructor and requires the use of setters for readability.
-/// e.g.
-/// Readable:     [SKFunctionInput(Description = "...", DefaultValue = "...")]
-/// Not readable: [SKFunctionInput("...", "...")]
+/// The attribute may be applied to any string parameter in the function signature.
+/// It may be applied to at most one parameter in a function signature.
+/// The attribute allows providing a default value if no main Input is available.
 /// </remarks>
 /// <example>
 /// <code>
-///   // No main parameter here, only context
-///   public async Task WriteAsync(SKContext context
+/// // No main parameter here, only context
+/// public async Task WriteAsync(SKContext context)
 /// </code>
 /// </example>
 /// <example>
 /// <code>
-///   // "path" is the input parameter
-///   [SKFunctionInput("Source file path")]
-///   public async Task{string?} ReadAsync(string path, SKContext context
+/// // No main parameter; path parameter is looked up using the name "path"
+/// public async Task{string?} ReadAsync(string path, SKContext context)
 /// </code>
 /// </example>
-[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+/// <example>
+/// <code>
+/// // Main parameter; path parameter is looked up using the main name "input"
+/// public async Task{string?} ReadAsync([SKFunctionInput("Source file path")] string path, SKContext context)
+/// </code>
+/// </example>
+[AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
 public sealed class SKFunctionInputAttribute : Attribute
```

Comment:
Same as above - since we are changing the AttributeTargets nature of the attribute, we may want to take the opportunity to simplify the name, align with the Parameter version.
- SKFunctionInput (current)
- SKInput

Another argument for new names: we could feasibly leave the current attributes in place with "Obsolete" attributes of their own, directing developers to use the new ones (for one or two nuget drops, then remove). This will be a lot of churn, changes needed, so I'm sure they'll appreciate the upgrade assistance. (We'll also document in Release notes)